app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php

```diff
@@ -33,10 +35,13 @@ class AdaptAddIsInStockFilterToCollectionPlugin
      */
     public function __construct(
         GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite,
-        AddIsInStockFilterToCollection $addIsInStockFilterToCollection
+        AddIsInStockFilterToCollection $addIsInStockFilterToCollection,
+        DefaultStockProviderInterface $defaultStockProvider = null
     ) {
         $this->getStockIdForCurrentWebsite = $getStockIdForCurrentWebsite;
         $this->addIsInStockFilterToCollection = $addIsInStockFilterToCollection;
+        $this->defaultStockProvider = $defaultStockProvider ?: ObjectManager::getInstance()
```

Comment:
This class is a plugin and therefore is not covered by the backward compatibility development policies. There is no need to inject additional dependencies this way